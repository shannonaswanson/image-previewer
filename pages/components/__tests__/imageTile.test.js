import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import ImageTile from '../imageTile';

describe('ImageTile ', () => {
    const name = 'hello';
    const src = 'world';
    afterEach(() => cleanup());

    test('should render without crashing', () => {
        render(<ImageTile />);
    });

    test('should match snapshot', () => {
        const jsonTree = renderer.create(<ImageTile name={name} src={src} />).toJSON();
        expect(jsonTree).toMatchSnapshot();
    });

    test('should render with name and src attribute', () => {
        render(<ImageTile name={name} src={src} />);
        const imageTileElement = screen.getByTestId('image-tile');
        expect(imageTileElement).toHaveTextContent(name);

        const imageContainer = screen.getByTestId('image-container');
        expect(imageContainer).toHaveAttribute('src', src);
    });
});