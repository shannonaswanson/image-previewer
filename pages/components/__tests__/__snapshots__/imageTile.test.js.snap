// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageTile  should match snapshot 1`] = `
<div
  data-testid="image-tile"
>
  <h3
    className="h3__ImageTitle-sc-2ay7y7-0 hwAtaM"
    data-testid="image-title"
  >
    hello
  </h3>
  <img
    className="imageContainer__ImageContainer-sc-139ry6z-0 cvbhVk"
    data-testid="image-container"
    src="world"
  />
</div>
`;
